source-code/plugins/json/src/settings.ts

```diff
@@ -1,47 +1,45 @@
+// pluginOptions for i18next plugin
+export type PluginSettings = {
+	pathPattern: string | Record<string, string>
+	variableReferencePattern?: string[] //default is ["{{", "}}"]
+	ignore?: string[] // default is []
+}
+
 /**
- * Throws an error if the settings are invalid.
+ * Throws an error if the options are invalid.
  *
  * Not using zod becaue it's not worth the bundle size (2kb vs 14kb).
  */
 export function throwIfInvalidSettings(settings: PluginSettings) {
```

Comment:
I think it makes sense to use typebox here too

Reply:
@ivanhofer you should ping @inlang/plugins if you want a reply ;)

Reply:
I will add it 👍

Reply:
To be honest, I have troubles understanding the current validation logic. Some things get validated outside this function and something inside the function looks like code duplication.
It was just a suggestion, to have the same validatin pattern everywhere.